src/validate-params.ts

```diff
@@ -2,18 +2,22 @@ import {joinErrors, toResult, Result} from './util';
 import * as chalk from 'chalk';
 import SoyContext from './soy-context';
 import JSContext from './js-context';
-import {Config} from './config';
+import {Config, ImplicitParamsMap} from './config';
 
-function getImplicitParams(className: string | null, paramsMap: any):string[] {
-  return className ? Object.keys(paramsMap)
-    .filter(regex => className.match(new RegExp(regex)))
-    .reduce((acc, value) => acc.concat(paramsMap[value]), []) : [];
+function getImplicitParams(className: string, paramsMap: ImplicitParamsMap): Array<string> {
+  const initialImplicitParams: Array<string> = [];
```

Comment:
This seems to be needed so `reduce` gets the proper signature. Otherwise, it defaults to `never[]`, and you can't `never[].concat(string|string[])`. Makes sense?